src/cephadm/cephadm

```diff
@@ -2370,6 +2401,12 @@ def command_adopt():
         call_throws([container_path, 'pull', args.image])
 
     (daemon_type, daemon_id) = args.name.split('.', 1)
+
+    if daemon_type == 'prometheus':
+        (uid, gid) = extract_uid_gid_monitoring(daemon_type)
```

Comment:
If we could call `extract_uid_gid` here instead, would it be possible to just continue with the flow, till after the `l.acquire()` call?